soh/soh/Enhancements/randomizer/ShufflePots.cpp

```diff
@@ -0,0 +1,126 @@
+        (play->sceneNum > SCENE_TREASURE_BOX_SHOP && play->sceneNum < SCENE_GANONS_TOWER_COLLAPSE_EXTERIOR);
+    uint8_t potSetting = Randomizer_GetSettingValue(RSK_SHUFFLE_POTS);
+
+    // Don't pull randomized item if pot isn't randomized or is already checked
+    if (!IS_RANDO || (potSetting == RO_SHUFFLE_POTS_OVERWORLD && isDungeon) ||
+        (potSetting == RO_SHUFFLE_POTS_DUNGEONS && !isDungeon) ||
+        Flags_GetRandomizerInf(potActor->potIdentity.randomizerInf) ||
+        potActor->potIdentity.randomizerCheck == RC_UNKNOWN_CHECK) {
+        return false;
+    } else {
+        return true;
+    }
+}
+
+void ObjTsubo_RandomizerSpawnCollectible(ObjTsubo* potActor, PlayState* play) {
+    EnItem00* item00 = (EnItem00*)Item_DropCollectible2(play, &potActor->actor.world.pos, ITEM00_SOH_DUMMY);
+    item00->randoInf = potActor->potIdentity.randomizerInf;
+    item00->itemEntry =
+        OTRGlobals::Instance->gRandomizer->GetItemFromKnownCheck(potActor->potIdentity.randomizerCheck, GI_NONE);
+    item00->actor.draw = (ActorFunc)EnItem00_DrawRandomizedItem;
+    item00->actor.velocity.y = 8.0f;
+    item00->actor.speedXZ = 2.0f;
+    item00->actor.world.rot.y = Rand_CenteredFloat(65536.0f);
+}
+
+void ObjTsubo_RandomizerInit(void* actorRef) {
+    Actor* actor = static_cast<Actor*>(actorRef);
+
+    if (actor->id != ACTOR_OBJ_TSUBO) return;
+
+    ObjTsubo* potActor = static_cast<ObjTsubo*>(actorRef);
+
+    potActor->potIdentity =
+        Randomizer_IdentifyPot(gPlayState->sceneNum, (s16)actor->world.pos.x, (s16)actor->world.pos.z);
+}
+
+void ShufflePots_OnVanillaBehaviorHandler(GIVanillaBehavior id, bool* should, va_list originalArgs) {
+    va_list args;
+    va_copy(args, originalArgs);
+
+    // Draw custom model for pot to indicate it holding a randomized item.
+    if (id == VB_POT_DRAW) {
+        ObjTsubo* potActor = va_arg(args, ObjTsubo*);
+        if (ObjTsubo_RandomizerHoldsItem(potActor, gPlayState)) {
+            potActor->actor.draw = (ActorFunc)ObjTsubo_RandomizerDraw;
+            *should = false;
+        }
+    }
+
+    // Do not spawn vanilla item from pot, instead spawn the ranomized item.
+    if (id == VB_POT_DROP_ITEM) {
+        ObjTsubo* potActor = va_arg(args, ObjTsubo*);
+        if (ObjTsubo_RandomizerHoldsItem(potActor, gPlayState)) {
+            ObjTsubo_RandomizerSpawnCollectible(potActor, gPlayState);
+            *should = false;
+        }
+    }
+
+    // Unlock early Ganon's Boss Key doors to allow access to the pots there when pots are shuffled in dungeon
+    if (id == VB_LOCK_DOOR) {
+        DoorShutter* doorActor = va_arg(args, DoorShutter*);
+        uint8_t shufflePotSetting = Randomizer_GetSettingValue(RSK_SHUFFLE_POTS);
+        if (gPlayState->sceneNum == SCENE_GANONS_TOWER && doorActor->dyna.actor.world.pos.y == 800 &&
+            (shufflePotSetting == RO_SHUFFLE_POTS_DUNGEONS || shufflePotSetting == RO_SHUFFLE_POTS_ALL)) {
+            *should = false;
+        }
+    }
```

Comment:
I don't think this should be a thing (disregarding no logic) as imo it's more interesting to have checks behind gbk (which iirc is completely new) rather than just being able to access them earlier.
Something else to note is that logic doesn't account for this and still counts them as behind gbk.

Reply:
This coincides with how OoTR handles the boss door with Potsanity enabled. Its a generally desired approach to move the door forward and allow access to 18 more pots rather than forcibly block them behind GBK.

Reply:
I think it'd be best to make it a toggle for people that like either approach.

Reply:
I think this could be part of a dungeon shortcuts push at some point, but that's probably a while off because shortcuts in shadow enable annoying quantum key logic, so a small tickbox for now is fine.

Reply:
I've fixed the logic to no require the Ganon's Boss Key by introducing a new region for that area. It got changed down the line which I failed to catch.

That said, personally, I'm just not interested in supporting them being behind the boss key. In many settings, GBK is basically used as the go-mode or close to it, so not having access to these pots in logic means they'll just almost never be required, and in no logic it's basically 18 dead checks.

In triforce hunt they're also going to be completely dead locations.

If someone wants to implement the option down the line, go for it, but I'm going to be honest here, I don't want to.

Reply:
I think it's interesting as it adds the possibility of ms or the light arrows being behind gbk (maybe even by multiple steps), which is not possible otherwise. Regarding triforce hunt, we could change it so gbk is not given after winning and instead making it just a regular item with ganondorf/ganon not appearing (same as with boss souls) until you win by collecting the triforce pieces. In no logic they are dead checks unless there is a void warp that would work for this, which is unlikely; we could disable the option for no logic.
It's obviously not mandatory for you to implement it here but it's worth noting that there are legitimate reasons for wanting this option.